src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/DSA.cs

```diff
@@ -222,5 +231,261 @@ public virtual bool VerifyData(ReadOnlySpan<byte> data, ReadOnlySpan<byte> signa
             new ArgumentException(SR.Cryptography_HashAlgorithmNameNullOrEmpty, "hashAlgorithm");
+
+        public override bool TryExportEncryptedPkcs8PrivateKey(
+            ReadOnlySpan<byte> passwordBytes,
+            PbeParameters pbeParameters,
+            Span<byte> destination,
+            out int bytesWritten)
+        {
+            if (pbeParameters == null)
+                throw new ArgumentNullException(nameof(pbeParameters));
+
+            PasswordBasedEncryption.ValidatePbeParameters(
+                pbeParameters,
+                ReadOnlySpan<char>.Empty,
+                passwordBytes);
+
+            using (AsnWriter pkcs8PrivateKey = WritePkcs8())
+            using (AsnWriter writer = KeyFormatHelper.WriteEncryptedPkcs8(
+                passwordBytes,
+                pkcs8PrivateKey,
+                pbeParameters))
+            {
+                return writer.TryEncode(destination, out bytesWritten);
+            }
+        }
+
+        public override bool TryExportEncryptedPkcs8PrivateKey(
+            ReadOnlySpan<char> password,
+            PbeParameters pbeParameters,
+            Span<byte> destination,
+            out int bytesWritten)
+        {
+            if (pbeParameters == null)
+                throw new ArgumentNullException(nameof(pbeParameters));
+
+            PasswordBasedEncryption.ValidatePbeParameters(
+                pbeParameters,
+                password,
+                ReadOnlySpan<byte>.Empty);
+
+            using (AsnWriter pkcs8PrivateKey = WritePkcs8())
+            using (AsnWriter writer = KeyFormatHelper.WriteEncryptedPkcs8(
+                password,
+                pkcs8PrivateKey,
+                pbeParameters))
+            {
+                return writer.TryEncode(destination, out bytesWritten);
+            }
+        }
+
+        public override bool TryExportPkcs8PrivateKey(
+            Span<byte> destination,
+            out int bytesWritten)
+        {
+            using (AsnWriter writer = WritePkcs8())
+            {
+                return writer.TryEncode(destination, out bytesWritten);
+            }
+        }
+
+        public override bool TryExportSubjectPublicKeyInfo(
+            Span<byte> destination,
+            out int bytesWritten)
+        {
+            using (AsnWriter writer = WriteSubjectPublicKeyInfo())
+            {
+                return writer.TryEncode(destination, out bytesWritten);
+            }
+        }
+
+        private AsnWriter WritePkcs8()
+        {
+            DSAParameters dsaParameters = ExportParameters(true);
+
+            AsnWriter writer = new AsnWriter(AsnEncodingRules.DER);
+
+            writer.PushSequence();
+            writer.WriteInteger(0);
+            WriteAlgorithmId(writer, dsaParameters);
+            WriteKeyComponent(writer, dsaParameters.X, bitString: false);
+            writer.PopSequence();
+
+            CryptographicOperations.ZeroMemory(dsaParameters.X);
```

Comment:
How important is this clearing?  Does it need to be a finally in case, for example, we OOM while allocating AsnWriter?  Also, is the writer dsaParameters.X copied to pinning / clearing the data?  I'm generally wondering about all this clearing we're doing, and whether we're actually solving a problem.